vibra/interface/general/selection_handler.py

```diff
@@ -0,0 +1,122 @@
+from typing import TYPE_CHECKING
+from PySide6.QtCore import Signal, QObject
+
+if TYPE_CHECKING:
+    from vibra.engine.mesher.mesh import Mesh
+
+class SelectionHandler(QObject):
+    selection_changed = Signal()
+    
+    def __init__(self, mesh: "Mesh | None" = None):
```

Comment:
```suggestion
    def __init__(self, mesh: Mesh | None = None):
```

It works without quotes if you add `from __future__ import annotations` on the imports.
Both options are a bit ugly, but importing from future improves the code highlighting (at least in my editor).